spring-integration-core/src/main/java/org/springframework/integration/dsl/IntegrationFlowBeanPostProcessor.java

```diff
@@ -252,9 +255,33 @@ else if (component instanceof AnnotationGatewayProxyFactoryBean) {
 
 	private Object processIntegrationFlowImpl(IntegrationFlow flow, String beanName) {
 		IntegrationFlowBuilder flowBuilder = IntegrationFlows.from(beanName + ".input");
+
 		flow.configure(flowBuilder);
-		Object standardIntegrationFlow = processStandardIntegrationFlow(flowBuilder.get(), beanName);
-		return isLambda(flow) ? standardIntegrationFlow : flow;
+
+		StandardIntegrationFlow target = flowBuilder.get();
+		processStandardIntegrationFlow(target, beanName);
+
+		if (!(flow instanceof IntegrationFlowAdapter)) {
+			NameMatchMethodPointcutAdvisor integrationFlowAdvice =
+					new NameMatchMethodPointcutAdvisor(new IntegrationFlowLifecycleAdvice(target));
+			integrationFlowAdvice.setMappedNames(
+					"getInputChannel",
+					"start",
+					"stop",
+					"isRunning",
+					"isAutoStartup",
+					"getPhase");
+
+			ProxyFactory proxyFactory = new ProxyFactory(flow);
+			proxyFactory.addAdvisor(integrationFlowAdvice);
+			if (!(flow instanceof SmartLifecycle)) {
+				proxyFactory.addInterface(SmartLifecycle.class);
```

Comment:
Suggested method comments here https://github.com/garyrussell/spring-integration/commit/ee8015a87998106f2c4ae5e5d5a7fae8102bd2b2

I wonder if we should use `ClassUtils.getAllInterfaces`, and add them here, in case a user implements `IntegrationFlow` and additional interfaces.

Reply:
We don't need that information. The `Advice` is applied only to the restricted set of methods. Everything else is left untouched.

We add a `SmartLifecycle` here explicitly because we need it for the Manual flow registration and really for propagation lifecycle control to the `target`, which is not available from the application context anymore.

Reply:
Interesting "why?". This is JDK Proxy, so all original interfaces must be on the target wrapped object...

Reply:
```
public interface Foo {

}

@Component
public static class MyFlow implements IntegrationFlow, Foo {
```

and in the test case:
```
assertThat(this.myFlow, instanceOf(Foo.class));
```

What am I missing?

Reply:
I also have just added `SmartLifecycle` into that `MyFlow` class definition. Works well.
:man_shrugging:

Reply:
Doh; yes; of course; I was thinking CGLib.